### DISCUSSION TITLE
API endpoint requests

### DISCUSSION THREAD

**Discussion by hughrawlinson** [BattlesnakeOfficial/feedback#92] [Feature Requests]
I have a load of API related feature requests that I thought I would dump as one issue and be happy to split into separate issues if you think they're worth having a discussion about.

### Customizations API

I'm working on a battlesnake generator, and when a developer is generating a battlesnake, right now they need to manually type in the name of the head and tail they would like to use. If we had an API, I could give the users a menu, and maybe even display the image of the head/tail to them.

### Oauth API

It would be cool if I could let battlesnake developers log in to my battlesnake generator/tooling so that it could act on their behalf for actions on the website such as:

### Snake registration/configuration API

My generated snakes have ngrok built in for local development, and right now the developers have to copy the ngrok url and paste it into the battlesnake website. If they're already logged into my tool with oauth, it would be great to be able to automatically match their local snake to one of their registered snakes (or allow them to register a new snake) and update the URL for oauth.

### Game triggering API

At the moment an hypothetical snake development loop is:

- go to the battlesnake site and start a game with my snake and some opponents
- watch what my snake did
- stop my snake
- modify my code to account for what my snake did
- start my snake
- update my ngrok url
- repeat

It would be cool to be able to have my tooling use a battlesnake API to start a game when the snake server runs.

I realize that there are ways to keep ngrok tunnels open longer, for sure, but still.

---

Any thoughts on those?

**Comment by bvanvugt** [comment]
@hughrawlinson love this, thanks for putting it together. I'm going to parse through it this week, probably live on stream, and see what we can do 👍

**Reply by hughrawlinson** [reply]
Sounds good! I'll try to be in Twitch chat :)
